File: flutter_app4/android/app/src/main/kotlin/com/pan/flutter_app4/method_channels/FlutterMethodChannels.kt
package com.pan.flutter_app4.method_channels

import android.util.Log
import io.flutter.app.FlutterActivity
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel
import java.util.*


object FlutterMethodChannels {

    private var registered = false;

    init {

    }

    fun registerWith(flutterActivity: FlutterActivity) {
        if(registered){
            return
        }
        try {


            val channel = MethodChannel(flutterActivity.flutterView, "pan.flutter.dev/default_method_channel");

            channel.setMethodCallHandler { call, result ->
                if(call.method == "getDebugMsg"){


                    val arguments = call.arguments as? Map<String, Any>
                    arguments?.forEach { (k, v) ->
                        Log.d("TAG", "entry = ${k}, ${v}")
                    }



                    channel.invokeMethod("getDartNumber", null, object : MethodChannel.Result {
                        override fun notImplemented() {

                        }

                        override fun error(p0: String?, p1: String?, p2: Any?) {
                            Log.d("TAG", "error = ${p1}, ${p2}")

                        }

                        override fun success(p0: Any?) {
                            Log.d("TAG", "success = ${p0}")

                        }
                    })


                    result.success("Msg from Android: received = ${Date()}, call arg = ${arguments}")
                }
            }

        } finally {
            registered = true
        }
    }

}

File: flutter_app4/ios/Runner/FlutterMethodChannels.swift
//
//  FlutterMethodChannels.swift
//  Runner
//
//  Created by Ng Yat Pan on 10/6/19.
//  Copyright © 2019 The Chromium Authors. All rights reserved.
//

import Foundation

class FlutterMethodChannels {
    static let shared = FlutterMethodChannels()
    
    init(){
        
    }
    
    func register(with: FlutterAppDelegate){
        guard let fvc = with.window.rootViewController as? FlutterViewController else {
            return
        }
        
        
        let channel = FlutterMethodChannel(name: "pan.flutter.dev/default_method_channel",
                                                       binaryMessenger: fvc)
        
        
    
        
        channel.setMethodCallHandler({
            (call: FlutterMethodCall, result:@escaping FlutterResult) -> Void in
            switch call.method {
            case "getDebugMsg":
               
                
                channel.invokeMethod("getDartNumber", arguments:  nil, result : {
                    dartResult in
                    if let successResult = dartResult as? NSNumber {
                        result("Success iOS: \(successResult)")
                        NSLog("Success iOS \(successResult)")
                    } else {
                        result("from iOS: 1234")
                        NSLog("Error iOS")
                    }
                });
                
                
                 NSLog("from iOS: 1234")
               //  result("from iOS: 1234")
                
            default:
                result(FlutterMethodNotImplemented)
            }
        })
        
        FlutterMethodChannel.method
    }
}
